src/swap/defi/uni-v2-based/plugins/spookySwap.ts

```diff
@@ -41,6 +41,10 @@ export function makeSpookySwapPlugin(
   const { quiknodeApiKey } = asInitOptions(opts.initOptions)
   const provider = getFtmProvider(quiknodeApiKey)
 
+  // For max swaps, save the gas price from the first getSwapTransactions call and pass it into the second
+  let isMaxSwap = false
+  let previousGasPrice: string | undefined
```

Comment:
Using these external variables could cause a race condition if somebody requests multiple swaps in parallel. I would like to see the previous version of this PR, so we can kit-bash those changes with these ones to get a single stable solution.

Reply:
The previous solution was to pipe the custom network fee from the first makeSpend (in getMaxSwappable) and return it with the newRequest back to the swap plugin. Then it would be passed it into the inner function with the request and other args. 

That solution is better and would be a step towards resolving another race condition where calling two makeSpends, without custom fees, can result in different fees for each one. We don't currently have a way to grab the fee from a makeSpend (feeRateUsed is for display and doesn't map perfectly for all currency plugins) but if we did we could grab it make sure it's included in the second go around with the max amount. 

For the solution that's currently in the PR, I could make utility function that holds a uid for each request that the plugins can access. I'll add that as a fixup

Reply:
Added a fixup that resolves the race condition using a map at a higher scope